### PR TITLE
Fix to dumbbell initialization in layer mode

### PR DESCRIPTION
**Pull request by WenhaoChen89**
This PR fixes the initialization of the dumbbell test to run with the layer mode. 
1. The initial layer thickness is changed to follow the isopycnal
2. The initial salinity is set up using the 'fit' method. 
3. The sponge is added for the layer mode

### CODE REVIEW

File: src/user/dumbbell_initialization.F90 [hunk 1]
```diff
@@ -128,6 +132,8 @@ subroutine dumbbell_initialize_thickness ( h, depth_tot, G, GV, US, param_file,
                  units='m', default=1.0e-3, scale=US%m_to_Z, do_not_log=just_read)
   call get_param(param_file, mdl,"REGRIDDING_COORDINATE_MODE", verticalCoordinate, &
                  default=DEFAULT_COORDINATE_MODE, do_not_log=just_read)
+  call get_param(param_file, mdl, "USE_REGRIDDING", use_ALE, do_not_log = .true.)
+  if(.not. use_ALE) verticalCoordinate = "LAYER"
```
Comment (Hallberg-NOAA) [inline src/user/dumbbell_initialization.F90:136]: Please add a space between if and the following parenthesis, as discussed in the MOM6 style guide at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#white-space.

File: src/user/dumbbell_initialization.F90 [hunk 2]
```diff
@@ -231,12 +265,18 @@ subroutine dumbbell_initialize_temperature_salinity ( T, S, h, G, GV, US, param_
...
+  logical :: use_ALE     ! If false, use layer mode.
   character(len=20) :: verticalCoordinate, density_profile
 
   is = G%isc ; ie = G%iec ; js = G%jsc ; je = G%jec ; nz = GV%ke
 
   T_surf = 20.0*US%degC_to_C
 
+  ! layer mode
+  call get_param(param_file, mdl, "USE_REGRIDDING", use_ALE, do_not_log = .true.)
+  if(.not. use_ALE) call MOM_error(FATAL,  "dumbbell_initialize_temperature_salinity: "//&
```
Comment (Hallberg-NOAA) [inline src/user/dumbbell_initialization.F90:277]: As with line 136, please add a space between if and the following parenthesis, as discussed in the MOM6 style guide at https://github.com/NOAA-GFDL/MOM6/wiki/Code-style-guide#white-space.

**Review by Hallberg-NOAA — CHANGES_REQUESTED** [review]
Overall this is a useful contribution that is moving in the right direction, but there is a dimensionally incorrect expression (noted in the line-specific comments) that will need to be corrected before this PR can be accept.  As written, this would fail the dimensional consistency testing if Z_RESCALE_POWER is not set equal to H_RESCALE_POWER.

File: src/user/dumbbell_initialization.F90 [hunk 3]
```diff
@@ -404,9 +446,26 @@ subroutine dumbbell_initialize_sponges(G, GV, US, tv, depth_tot, param_file, use
...
       endif
     enddo ; enddo
-  endif
-
   if (associated(tv%S)) call set_up_ALE_sponge_field(S, G, GV, tv%S, ACSp)
+ else
+    do j=G%jsc,G%jec ; do i=G%isc,G%iec
+      eta(i,j,1) = 0.0
+      do k=2,nz
+        eta(i,j,k) = eta(i,j,k-1)-h_in(i,j,k-1)
```
Comment (Hallberg-NOAA) [inline src/user/dumbbell_initialization.F90:454]: This expression calculating the interface heights is dimensionally inconsistent (eta and depth_tot have units of Z, while h_in has units of H). This expression could be compared with the dimensionally correct one at line 175, above.

**Review by Hallberg-NOAA — COMMENTED** [review]
Some of the issues with this code have now been addressed, but the dimensionally inconsistent expression on line 454 of dumbbell_initialization.F90 still needs to be corrected by multiplying h_in by GV%H_to_Z.  I would also request that after this change is made, that this contribution should be tested for dimensional consistency by running it with various settings for H_RESCALE_POWER and Z_RESCALE_POWER; the answers should be identical for any values between about -140 and 140 for these factors. For completeness of testing these two parameters should NOT use the same value.

**Review by Hallberg-NOAA — APPROVED** [review]

**Comment by Hallberg-NOAA** [comment]
All of my concerns with this PR have now been addressed, and I see this as a valuable new contribution.

  I am, however, recommending that we handle this PR via a squash merge, as it has a relatively high number of commits (9) for a straightforward new contribution, and several of the commits are partial or full reversions of the earlier commits.

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/MOM6/-/pipelines/16147 :heavy_check_mark: 

(NOTE: There were multiple errors from the CI, but appear to be network-related and not to do with this PR.)

**Pull request merged**